html/browsers/the-window-object/closed-attribute.window.js

```diff
@@ -0,0 +1,67 @@
+// META: script=/common/get-host-info.sub.js
+
+function closedTest(newWindow, closeNewWindowsBrowsingContext) {
+  assert_equals(newWindow.closed, false);
+  closeNewWindowsBrowsingContext();
+  assert_equals(newWindow.closed, true);
+}
+
+test(() => {
+  const frame = document.body.appendChild(document.createElement("iframe"));
+  closedTest(frame.contentWindow, () => frame.remove());
+}, "closed and same-origin nested browsing context");
+
+test(() => {
+  const openee = window.open();
+  closedTest(openee, () => openee.close());
+
+  // This should not matter
+  openee.close();
+  assert_equals(openee.closed, true);
+}, "closed/close() and same-origin auxiliary browsing context");
+
+const support = new URL("support/closed.html", location.href).pathname;
+[
+  {
+    type: "cross-origin",
+    url: `${get_host_info().HTTP_REMOTE_ORIGIN}${support}`
+  },
+  {
+    type: "cross-site",
+    url: `${get_host_info().HTTP_NOTSAMESITE_ORIGIN}${support}`
+  }
+].forEach(val => {
+  async_test(t => {
```

Comment:
I guess you could make it more self-documenting by having some wrapper so that it was

```js
runTestWhileWaitingForLoad({
  type: "cross-origin",
  setup() {
    const frame = document.createElement("iframe");
    frame.src = url;
    document.body.append(frame);
  },
  test() {
    closedTest(frame.contentWindow, () => frame.remove());
    self.removeEventListener("message", listener);
  }
});
```

or something similar.